lib/mix/lib/mix/task.ex

```diff
@@ -428,6 +428,7 @@ defmodule Mix.Task do
     recursive = (module = get(task)) && recursive(module)
 
     Mix.TasksServer.delete_many([{:task, task, proj}, {:alias, task, proj}])
+    Mix.TasksServer.delete_task_for_all_projects({:task, task})
```

Comment:
I am afraid this change will have other consequences, as I really don't expect reenabling to work across all projects. In particular, we want to reenable only the local projects, and not all of the dependencies.

I am afraid the best way to solve this issue is to add a function like: `Mix.Project.within_local_projects(fn project_config -> ... end)` which will execute within all local projects, including the current project itself, and then we run `compile` inside of each them.

However, implementing this task is far from trivial, so I am currently inclined to mark the original issue as unresolved. :/

Thank you very much for looking into this! As we can only know how hard it is to provide it once we give it a try.